deploy/helm/values.yaml

```diff
@@ -591,7 +592,7 @@ swoagent:
   enabled: false
   image:
     repository: solarwinds/swo-agent
-    tag: ""
+    tag: "v2.6.28"
```

Comment:
I am ok with defining tag in values for images like busybox etc, but I am not OK defining it for images which we need to upgrade regularly, which includes swo-agent. This will make very hard to upgrade those images with upgrade of helm charts, because many users simply copy whole `values.yaml` at the time of install, therefore they would be fixed to specific version of the image even after upgrade of Helm. 

> > Moved default values for referenced containers from templates to the values.yaml. It's more in line with where default values should be specified. It's also useful for documenting how to change all these images, if needed.

Do you have some reference material supporting this statement? What I see in e.g. Splung helm is that they use similar pattern for images they own (not all though): https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/values.yaml#L957

Reply:
This is a complicated topic.

From https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function:
> In an actual chart, all static default values should live in the values.yaml, and should not be repeated using the default command (otherwise they would be redundant). However, the default command is perfect for computed values, which cannot be declared inside values.yaml.

And Helm community has several discussions on this, e.g. `https://github.com/helm/helm/issues/7754`.
But the current "standard" seems to be that the main image should be using the `.Chart.AppVersion` by default, while 3rd party image tags are explicitly listed in `values.yaml`, sometimes as specific versions, sometimes as `latest`. As an example, here is such setting in the Splunk Helm, that you referenced: https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/values.yaml#L949.

Do we consider the UAMS agent a 3rd party image to the k8s collector?

And regarding copying the whole file by customers and using it just a bit modified, well, that can happen. But does it really? The file is getting bigger and bigger. And now with schema validation, they should have even fewer reasons to not just write their own overrides.

Reply:
Yes, we used to have such concerns. But that was when the default file had just a few dozen lines, at most. Today, I doubt that anyone would just copy it.